gradient_estimators/random_gradient_estimator.py

```diff
@@ -86,11 +88,13 @@ def _forward_method(self, batch_inputs, labels, criterion):
             self.perturb_model(pb_norm, alpha=-self.mu)  # Restore model
 
             dir_grad = (pert_plus_loss - initial_loss) / self.mu
+            dir_grads += [dir_grad]
             grad += dir_grad * pb_norm
-        return grad / self.num_pert
+        return grad / self.num_pert, torch.tensor(dir_grads, device=self.device)
```

Comment:
I don't think this is right. It should be `torch.cat(dir_grads, dim=0)`

Reply:
It's doing what it's supposed to do. See following screenshot for more details.

dir_grads is a list of scalar tensor. Each elements in dir_grads is a scalar tensor, whose shape is [].
Directly use torch.tensor on it creates a 1-d tensor.


![image](https://github.com/ZidongLiu/CeZO-FL/assets/26914775/fab9a067-50d5-429b-b540-74508e090bff)